test/qt3testsBenchmark.ts

```diff
@@ -0,0 +1,274 @@
+import { Suite } from 'benchmark';
+
+import {
+	evaluateXPathToBoolean,
+	evaluateXPathToFirstNode,
+	evaluateXPathToNodes,
+	evaluateXPathToNumber,
+	evaluateXPathToString,
+	Language,
+	Document,
+} from 'fontoxpath';
+
+import testFs from 'test-helpers/testFs';
+import { ALL_TESTS_QUERY, getAllTestSets, getArguments, getFile } from 'test-helpers/qt3TestsTools';
+import getPerformanceTests from 'test-helpers/getPerformanceTests';
+import { Node, Element } from 'slimdom';
+
+const suite: Suite = new Suite();
+
+function addTests(functions) {
+	Object.keys(functions).forEach((name) => {
+		suite.add(name, functions[name]);
+	});
+}
+
+function runTests() {
+	console.log('Tests are running...');
+	suite
+		// add listeners
+		.on('cycle', (event) => {
+			console.log(String(event.target));
+		})
+		.on('complete', () => {
+			testFs.writeFileSync(
+				'qt3PerformanceTestResults.csv',
+				suite
+					.map(
+						(e) =>
+							`${e.name},${Math.round(e.hz * 1000) / 1000} op/sec,±${
+								Math.round(e.stats.rme * 1000) / 1000
+							}%`
+					)
+					.join('\n')
+			);
+			console.log('Done!');
+		})
+		.run({ async: true });
+}
+
+function addTestCaseToBenchmark(
+	assertNode: Element,
+	language: Language,
+	xpath: string,
+	contextNode: Node | undefined,
+	variablesInScope: {},
+	namespaceResolver: (prefix: any) => any,
+	name: string
+) {
```

Comment:
NIT: The default return type of a function is `any` not `void`.
i.e. Please specify `: void` as return type.

Reply:
if I change it to `void`, `evaluateXPath` functions whines, because they expect a function returning `string` as `namesapceResolver`

Reply:
`namespaceResolver: (prefix: string) => string;` is this better?

Reply:
the actual thing should be `namespaceResolver?: (s: string) => string | null;`. From evaluateXPath.ts line 32

Reply:
Yeah that's better for the `namespaceResolver` indeed, I did it wrong when I send you that sample code over Slack.

I was referring to the function `addTestCaseToBenchmark` it now has no return type defined which defaults to `any` but it should be `void`. So add ` : void` right after `)` on this line.